app/models/vm_or_template.rb

```diff
@@ -393,7 +393,7 @@ def self.task_arguments(options)
     when "remove_snapshot", "revert_to_snapshot" then
       [options[:snap_selected]]
     when "create_snapshot" then
-      [options[:name], options[:description], options[:memory]]
+      [options[:name], options[:description], options.fetch(:memory, false)]
```

Comment:
@chessbyte @Fryguy 

on further discussions with Greg M we agreed to expose the memory parameter via the service model apis for vmware. This way the method api will enforce that the memory value can only be true|false.